### DISCUSSION TITLE
How to make one key stroke emit 3 unicode character in a custom keyboard layout in Linux mint?

### DISCUSSION THREAD

**Discussion by ritulahkar** [uim/uim#206] [Q&A]
I have created a custom keyboard layout for my regional language by editing different files. It works fine except for one problem. One of the characters needs 3 unicode characters in combination I.e. I want all the 3 characters to be emitted in single key stroke. Please help me to do so. 

The characters are as follows 

`ক + ্ + ষ --> ক্ষ`

`U0995 + U09CD + U09B7 --> ক্ষ`

This combination needs to be emitted from the third character of the key `key <AB04>`, shown below.

I added language file to `/usr/share/X11/xkb/symbols` and entry to the `/usr/share/X11/xkb/rules/evdev.xml` file

extract from the file I added:
```      
// shift, z to /
      // key <LFSH> { [ Shift_L ] };
               key <AB01> { [ U09A2,	U09A7,	U09B7,	voidsymbol ] };
               key <AB02> { [ U09A3,	U09A8,	U09B8,	voidsymbol  ] };
               key <AB03> { [ U09AA,	U09AF,	U09B9,	voidsymbol  ] };
               key <AB04> { [ U09AB,	U09F0,	voidsymbol,	voidsymbol ] };
               key <AB05> { [ U09AC,	U09B2,	U09DC,	voidsymbol ] };
               key <AB06> { [ U09AD,	U09F1,	U09DD,	voidsymbol ] };
               key <AB07> { [ U09AE,	U09B6,	U09DF,	voidsymbol ] };

```
Needs to add the letter ক্ষ to the 3 rd position of 6th key in the shift L line from left.

[![Needs to add the letter ক্ষ to the 3 rd position of 6th key in the shift L line from left.][1]][1]

I am using Linux mint 21.3.

  [1]: https://i.stack.imgur.com/653ee.png

**Comment by kou** [comment]
Hmm. It seems that this is not related to uim.

BTW, `/usr/share/X11/locale/en_US.UTF-8/Compose` may help you how to add a new symbol.

**Reply by ritulahkar** [reply]
Thank you for your reply. 
Can you provide a complete guidance for the same?

Actually, I asked the same question in stackoverflow.com. Someone advised that uim maybe of some help. But I could not find any good tutorials or guidance. That's why I asked here.

**Reply by kou** [reply]
I'm not familiar with it. Please read the file and files under `/usr/share/X11/xkb/`.

> Actually, I asked the same question in stackoverflow.com. Someone advised that uim maybe of some help. But I could not find any good tutorials or guidance. That's why I asked here.

In general, you should provide links to related questions. If you don't provide them, OSS maintainers and so on will waste their time.
